### PR TITLE
chore::> Lottery Factory ready

### PR DESCRIPTION
**Pull request by PsychoPunkSage**
Lottery Factory is Ready....
it has 3 components...
* `lotteryInterface` -> interface of `Factory` and `Handlers`
* `lotteryTokens` -> Handlers of different `NFTTypes`
* `lotteryFactory` -> Actual Factory

### CODE REVIEW

File: contracts/apps/factory/lotteryTokens.sol [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    function isApprovedForAll(address owner, address operator) external view returns (bool) {
+        return nftContract.isApprovedForAll(owner, operator);
+    }
+}
+
+contract ERC1155Handler is INFTHandler {
+    IERC1155 private nftContract;
+    uint256 private immutable tokenId;
```
Comment (pablin-dev) [inline contracts/apps/factory/lotteryTokens.sol:38]: A user can have multiple tokenId with multiple quantities.

File: contracts/apps/factory/lotteryTokens.sol [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    function getMaxSupply() external view returns (uint256) {
+        return maxSupply;
+    }
+
+    function ownerOf(uint256) external pure returns (address) {
+        // ERC1155 doesn't have direct ownerOf, would need additional tracking
+        revert ERC1155Handler__NotSupported();
+    }
+
+    function balanceOf(address user) external view returns (uint256) {
```
Comment (pablin-dev) [inline contracts/apps/factory/lotteryTokens.sol:62]: I think here we need the tokenId as parameter.

**Review by pablin-dev — CHANGES_REQUESTED** [review]
LGTM! Minor comments. I was thinking that to avoid external callbacks, maybe having only `one` owner for nfts,lottery,  proxy and factory could reduce the vector attacks by checking ownable on critical calls and only leave `claim()` and `claimNFT()` open.

File: contracts/apps/factory/lotteryTokens.sol (hunk 1 above)
Comment (PsychoPunkSage) [reply contracts/apps/factory/lotteryTokens.sol]: Done

File: contracts/apps/factory/lotteryTokens.sol (hunk 2 above)
Comment (PsychoPunkSage) [reply contracts/apps/factory/lotteryTokens.sol]: Done

**Comment by PsychoPunkSage** [comment]
@pablin-dev 
Made required changes

**Pull request merged**